ui/app/controllers/optimize.js

```diff
@@ -1,27 +1,41 @@
 import Controller from '@ember/controller';
-import { tracked } from '@glimmer/tracking';
-import { sort } from '@ember/object/computed';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { inject as controller } from '@ember/controller';
 import { task, timeout } from 'ember-concurrency';
-import Ember from 'ember';
 
 export default class OptimizeController extends Controller {
-  @tracked recommendationSummaryIndex = 0;
-
-  summarySorting = ['submitTime:desc'];
-  @sort('model', 'summarySorting') sortedSummaries;
+  @service router;
+  @controller('optimize/summary') summaryController;
 
   get activeRecommendationSummary() {
-    return this.sortedSummaries.objectAt(this.recommendationSummaryIndex);
+    const currentRoute = this.router.currentRoute;
+
+    if (currentRoute.name === 'optimize.summary') {
+      return this.summaryController.model;
+    } else {
+      return undefined;
+    }
```

Comment:
Hm. This feels weird at first read. A controller checking the current route name seems off. It looks like this is the parent route controller trying to also control behaviors on the child route?

Reply:
The purpose of the parent controller reaching into the the child is to determine the currently-active model in the summary table so it can:
* highlight the row in the table
* know where to go next when a summary is processed

Originally I had this replicating the same filtering that happens in the `optimize.summary` route: looking through the collection and matching against the stub and namespace. I didn’t love having that in two places so I changed it in 7a3c1df to use the result of that calculation as accessible through the child controller. I don’t love the top-down data access but I guess I love the duplication even less? haha

The route-checking was a defensive move that turned out to be unnecessary! I have removed it in 7e458e0b3 🎉